### PR TITLE
[docker] Update dependencies in dockerfile

### PR DESCRIPTION
**Pull request by xyang16**
## Description ##

Brief description of what this PR is about

- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

**Review by frankfliu — APPROVED** [review]

File: serving/docker/pytorch-gpu.Dockerfile [hunk 1]
```diff
@@ -44,7 +46,7 @@ ENV PYTORCH_FLAVOR=cu124-precxx11
 ENV TORCH_CUDNN_V8_API_DISABLED=1
 ENV JAVA_OPTS="-Xmx1g -Xms1g -XX:+ExitOnOutOfMemoryError -Dai.djl.default_engine=PyTorch"
 ENV HF_HOME=/tmp/.cache/huggingface
-ENV TRANSFORMERS_CACHE=/tmp/.cache/huggingface/transformers
```
Comment (sindhuvahinis) [inline serving/docker/pytorch-gpu.Dockerfile:47]: @xyang16 Don't we need TRANSFORMERS_CACHE anymore?

File: serving/docker/pytorch-gpu.Dockerfile [hunk 2]
```diff
@@ -21,6 +21,8 @@ ARG torch_vision_version=0.19.0
 ARG onnx_version=1.19.0
 ARG python_version=3.10
 ARG numpy_version=1.26.4
+ARG pydantic_version=2.8.2
+ARG djl_converter_wheel="https://publish.djl.ai/djl_converter/djl_converter-0.31.0-py3-none-any.whl"
```
Comment (sindhuvahinis) [inline serving/docker/pytorch-gpu.Dockerfile:25]: Curious, What is this for?

File: serving/docker/pytorch-gpu.Dockerfile (hunk 1 above)
Comment (frankfliu) [reply serving/docker/pytorch-gpu.Dockerfile:47]: This is deprecated, only need HF_HOME

File: serving/docker/pytorch-gpu.Dockerfile (hunk 2 above)
Comment (frankfliu) [reply serving/docker/pytorch-gpu.Dockerfile:25]: 1. `djl-converter` is used to convert huggingface model automatically to OnnxRuntime/Rust on the fly to optimize the performance
2. we depends on peft and pydantic when using `huggingface.py`, we should consider move them to where it used. right now they are hard dependency if run any python model even not related to peft in the container.

**Review by sindhuvahinis — APPROVED** [review]

**Pull request merged**